### PR TITLE
Downgrade required Qt6 version to 6.4

### PR DESCRIPTION
**Pull request by 375gnu**
Qt 6.5 is too new for the most of GNU/Linux users, e.g. Debian/Devuan stable has 6.4, the latest Ubuntu as well. It would be good to downgrade to 6.2 (packaged in Ubuntu LTS 22.04), but I have Devuan, so I cannot test with anything but 6.4.

### CODE REVIEW

**Comment by RacerBG** [comment]
Considering that the main LOOT package format for Linux is Flatpak I don't see a reason for downgrading. This would only make sense if someone is trying to build it from source. In that case, version 6.4 will be enough because 4 months from now Ubuntu 24.04 LTS will be released which will be based around Debian 12.

**Comment by Ortham** [comment]
I'm unwilling to downgrade the version of Qt used by official builds, but I think allowing LOOT to be built using older versions of Qt is fine. However, most of the the code this compiles conditionally should still be unconditionally compiled - I think the only bit of theming that relies on Qt 6.5 is the bit around the color scheme and detecting when it changes. All the LOOT theme settings should still be present when using Qt 6.4.

File: src/gui/qt/main_window.cpp [hunk 1]
```diff
@@ -326,6 +328,7 @@ void MainWindow::initialise() {
 }
 
 void MainWindow::applyTheme() {
+#if QT_VERSION >= QT_VERSION_CHECK(6, 5, 0)
```
Comment (Ortham) [inline src/gui/qt/main_window.cpp:331]: I think this `#if` should be moved to just before line 346 (the comment starting `Don't load default-dark`)

File: src/gui/qt/main_window.cpp [hunk 2]
```diff
@@ -269,7 +269,9 @@ MainWindow::MainWindow(LootState& state, QWidget* parent) :
 
 void MainWindow::initialise() {
   try {
+#if QT_VERSION >= QT_VERSION_CHECK(6, 5, 0)
```
Comment (Ortham) [inline src/gui/qt/main_window.cpp:272]: I don't think this is necessary.

File: src/gui/qt/main_window.cpp [hunk 3]
```diff
@@ -365,6 +368,7 @@ void MainWindow::applyTheme() {
         qobject_cast<CardDelegate*>(pluginCardsView->itemDelegate());
     cardDelegate->refreshStyling();
   }
+#endif
```
Comment (Ortham) [inline src/gui/qt/main_window.cpp:371]: I think this should be moved to before this `if` block.

File: src/gui/qt/settings/general_tab.cpp [hunk 4]
```diff
@@ -67,6 +67,7 @@ void GeneralTab::initialiseInputs(const LootSettings& settings,
       QVariant(QString::fromStdString(settings.getLanguage())));
   languageComboBox->setCurrentIndex(languageIndex);
 
+#if QT_VERSION >= QT_VERSION_CHECK(6, 5, 0)
```
Comment (Ortham) [inline src/gui/qt/settings/general_tab.cpp:70]: I don't think this is necessary.

File: src/gui/qt/settings/general_tab.cpp [hunk 5]
```diff
@@ -95,7 +97,9 @@ void GeneralTab::recordInputValues(LootSettings& settings) {
   auto defaultGame =
       defaultGameComboBox->currentData().toString().toStdString();
   auto language = languageComboBox->currentData().toString().toStdString();
+#if QT_VERSION >= QT_VERSION_CHECK(6, 5, 0)
```
Comment (Ortham) [inline src/gui/qt/settings/general_tab.cpp:100]: I don't think this is necessary.

File: src/gui/qt/settings/general_tab.cpp [hunk 6]
```diff
@@ -106,7 +110,9 @@ void GeneralTab::recordInputValues(LootSettings& settings) {
 
   settings.setDefaultGame(defaultGame);
   settings.setLanguage(language);
+#if QT_VERSION >= QT_VERSION_CHECK(6, 5, 0)
```
Comment (Ortham) [inline src/gui/qt/settings/general_tab.cpp:113]: I don't think this is necessary.

File: src/gui/qt/settings/general_tab.cpp [hunk 7]
```diff
@@ -135,7 +141,9 @@ void GeneralTab::setupUi() {
 
   generalLayout->addRow(defaultGameLabel, defaultGameComboBox);
   generalLayout->addRow(languageLabel, languageComboBox);
+#if QT_VERSION >= QT_VERSION_CHECK(6, 5, 0)
```
Comment (Ortham) [inline src/gui/qt/settings/general_tab.cpp:144]: I don't think this is necessary.

File: src/gui/qt/settings/general_tab.cpp [hunk 8]
```diff
@@ -151,7 +159,9 @@ void GeneralTab::setupUi() {
 void GeneralTab::translateUi() {
   defaultGameLabel->setText(translate("Default Game"));
   languageLabel->setText(translate("Language"));
+#if QT_VERSION >= QT_VERSION_CHECK(6, 5, 0)
```
Comment (Ortham) [inline src/gui/qt/settings/general_tab.cpp:162]: I don't think this is necessary.

File: src/gui/qt/settings/general_tab.h [hunk 9]
```diff
@@ -49,15 +49,19 @@ class GeneralTab : public QFrame {
 private:
   QLabel *defaultGameLabel{new QLabel(this)};
   QLabel *languageLabel{new QLabel(this)};
+#if QT_VERSION >= QT_VERSION_CHECK(6, 5, 0)
```
Comment (Ortham) [inline src/gui/qt/settings/general_tab.h:52]: I don't think this is necessary.

File: src/gui/qt/settings/general_tab.h [hunk 10]
```diff
@@ -49,15 +49,19 @@ class GeneralTab : public QFrame {
...
   QLabel *themeLabel{new QLabel(this)};
+#endif
   QLabel *updateMasterlistLabel{new QLabel(this)};
   QLabel *checkUpdatesLabel{new QLabel(this)};
   QLabel *loggingLabel{new QLabel(this)};
   QLabel *useNoSortingChangesDialogLabel{new QLabel(this)};
   QLabel *preludeSourceLabel{new QLabel(this)};
   QComboBox *defaultGameComboBox{new QComboBox(this)};
   QComboBox *languageComboBox{new QComboBox(this)};
+#if QT_VERSION >= QT_VERSION_CHECK(6, 5, 0)
```
Comment (Ortham) [inline src/gui/qt/settings/general_tab.h:62]: I don't think this is necessary.

File: src/gui/qt/style.cpp [hunk 11]
```diff
@@ -92,6 +92,9 @@ std::optional<QString> loadStyleSheet(const std::filesystem::path& themesPath,
 }
 
 std::vector<std::string> findThemes(const std::filesystem::path& themesPath) {
+#if QT_VERSION >= QT_VERSION_CHECK(6, 5, 0)
```
Comment (Ortham) [inline src/gui/qt/style.cpp:95]: I don't think this is necessary.

**Review by Ortham — CHANGES_REQUESTED** [review]

File: src/gui/qt/settings/general_tab.h (hunk 10 above)
Comment (375gnu) [reply src/gui/qt/settings/general_tab.h:62]: my idea was not to confuse users with settings which don't do anything, but guarding only `Qt::ColorScheme` is also fine.

Comment (Ortham) [reply src/gui/qt/settings/general_tab.h:62]: The theme settings predate the update to 6.5 though, they should work independently.

**Review by Ortham — APPROVED** [review]

**Pull request merged**